torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1476,16 +1489,47 @@ def _make_user_magic(method, user_type):
     else:
         method_attr = method
 
+    def get_constant(x: Union[SymInt, int, SymFloat, float, SymBool, bool]):
+        if isinstance(x, (int, float, bool)):
+            return x
+        if isinstance(x, SymBool):
+            return x.node.guard_bool("", 0)
+        raise AssertionError("expect to be called with constant SymBools")
+
+    def is_constant(x):
+        if isinstance(x, (int, float, bool)):
+            return True
+        if isinstance(x, (SymInt, SymFloat, SymBool)):
+            return x.node.is_constant()
+        return False
+
+    # Before performing the operation, check if any operands are constant.
+    # If so, extract out the constant values first. If `self` itself is a
+    # constant, then "redispatch" by calling back into the operator. Sometimes
+    # this means that operations involving SymBool return plain bools.
+    # Alternatively, we could also rewrap into constant Symbool (i.e. by
+    # implementing wrap_bool in ConstantSymNodeImpl), but we're not doing that
+    # today for no particular reason.
     def unary_magic_impl(self):
+        if is_constant(self):
+            return (method_to_operator(method))(get_constant(self))
         return wrap_node(getattr(self.node, method_attr)())
 
     def binary_magic_impl(self, other):
+        if is_constant(self):
+            return (method_to_operator(method))(get_constant(self), other)
```

Comment:
Does this handle other not being a constant correctly? 🤔

Reply:
Yeah, I think so - but let me know if you think I should test this case more deeply (we already do have a test, it just checks if the output is symbolic and guard_bool on that returns the expected value). What should happen is:
- If other is a symbolic SymBool, calling the operator with op(plain bool, symbolic SymBool) redispatches to the reflected op.
- At that point neither argument is constant and we proceed as usual (e.g. with symbolic SymBool promoting the "other" to another symbolic SymBool)

Reply:
oh you have a comment on this